### ISSUE TITLE
[CSV feed] CSV feed flood ingestion with identical data

### ISSUE THREAD

**Issue by Lhorus6** [LuatixHQ/opencti#8588]
## Description

CSV feed import seems buggy or not optimized.

In my case, I have an import from the Blocklist.de source, which contains around 30K IPs. E.g. at this moment, we have 27K entries in the source:

![image](https://github.com/user-attachments/assets/8db77f8f-42ac-457b-89fe-8b4195a902e8)

However, just for this small source, I currently find myself with 2.36M bundles in the queue and tons of works.

![image](https://github.com/user-attachments/assets/6302fc89-bd48-4c82-a466-7963aae9639e)

![image](https://github.com/user-attachments/assets/c08b73e9-7cc8-4fa6-87f7-7a76c1e25401)

![image](https://github.com/user-attachments/assets/0bcfed07-18ee-46b6-ad31-f06ba36b4c98)

## Environment

OCTI 6.3.4

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create this CSV Mapper:

![image](https://github.com/user-attachments/assets/43d8f5b1-b821-4450-ae05-812990adecd0)

2. Create this CSV feed: https://lists.blocklist.de/lists/all.txt

![image](https://github.com/user-attachments/assets/277146e9-c920-4222-a0f7-7b02ae0a4ed0)

3. Let it run for several hours, or even 24 - 48 hours, to see how it behaves.

## Additional information

It seems to me that it only imports the data if the hash changes. So this source updates its file every 30 minutes? (because I have a work every 30min)

This seems unlikely, perhaps we have a bug in the hash generation that takes meta data as input?
Just a guess

If the file does change continuously, maybe we shouldn't have to retrieve it every time, but just 2 times a day?

**Comment by nino-filigran** [comment]
I've started a feed to reproduce, will let you know about the output

**Comment by richard-julien** [comment]
We compute the hash on the full file.
We cant really do much on term of data control.
To prevent too much works I currently try to not create any job is there is something already in the queue

**Comment by nino-filigran** [comment]
I reproduced your issue @Lhorus6. So based on your comment @richard-julien , can I consider it as a "wont fix"?

**Comment by Lhorus6** [comment]
Maybe it's not the hash calculation we have to play with, but there's something to be done in any case IMO. Here we're blowing up the ingestion queues

Julien said " To prevent too much works I currently try to not create any job is there is something already in the queue", so I guess he is testing possibilities for improvement

**Comment by richard-julien** [comment]
Yes. Testing PR opened here https://github.com/OpenCTI-Platform/opencti/pull/8617

**Comment by Lhorus6** [comment]
Note: I think this issue is resolved by Julien's work. We tested a feed that was problematic and it's all good now.
